scoretracker/tracker.py

```diff
@@ -1,95 +1,164 @@
+                   disappointment: 420
+    @ed> /score ron disappointment + 420
+    @ed> /score show ron disappointment
+    @scorebot> ron:
+                   disappointment: 840
+    """
+    database = get_db()
+
+    database[person][score] += number
+
+    with open(DB_FILE, 'w') as db:
+        json.dump(database, db)
+
+    response = {
+        'bot_id': BOT_ID,
+        'text': 'Incremented {} score by {} for {}'.format(score, number, person)
+    }
+    requests.post(POST_URL, json.dumps(response))
+
+
+def get_scores(database, person, multi=True):
+    """
+    Return a string representing the scores a person has in the database.
+
+    The parameter `multi' is to specify whether the scores are used for
+    displaying the scores of a single person or multiple people.
+    """
+    indent = '     ' if multi else ''
+    return '\n'.join(
+        indent + '{score}: {points}'.format(score=score, points=points)
+        for score, points in database[person]
+    )
 
 
 def show_score(person=None, score=None):
-    rT = {}
-    rT['bot_id'] = BOT_ID
-    rT['text'] = ''
-    try:
-        with open(DB_FILE, 'r') as db:
-            try:
-                data = db.read()
-                database = json.loads(data)
-                if len(database) == 0:
-                    raise Exception()
-            except Exception:
-                rT['text'] = 'There are no scores yet!'
-                requests.post('https://api.groupme.com/v3/bots/post',
-                              json.dumps(rT))
-                return
-            if person is not None:
-                rT['text'] += person + ':\n'
-                if score is not None:
-                    rT['text'] += '    '+score+': '+database[person][score]+'\n'
-                else:
-                    for sco in database[person].keys():
-                        rT['text'] += '    '+sco+': '+database[person][sco]+'\n'
-            else:
-                for per, scores in database.items():
-                    rT['text'] += person + ':\n'
-                    for sco in scores:
-                        rT['text'] += '    '+sco+': '+database[per][sco]+'\n'
-    except Exception:
-        invalid_request()
-        return
-    requests.post('https://api.groupme.com/v3/bots/post', json.dumps(rT))
+    """
+    Sends to the chat the information about:
+    * All (user, score) combinations
+    * All scores of a given user
+    * A given user's score in a specific topic
+
+    @ed> /score show ron
+    @scorebot> ron:
+                   disappointment: 420
+                   understanding-the-rules: -69
+    """
+    response = {'bot_id': BOT_ID}
+    response['text'] = ''
+
+    database = get_db()
+
+    if not database:
+        # Backup did not exist or was empty
+        response['text'] = 'There are no scores yet!'
+
+    elif person is None:
+        response['text'] = '\n'.join(
+            '{person}:\n{scores}'.format(person=person, scores=get_scores(database, person))
```

Comment:
`get_scores` returns with a leading `\n` -- Get rid of it, replace with trailing `\n`.